horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -1004,4 +1010,188 @@ public DatasetList listDatasetsBySchema(String uri, Integer limit, Integer page,
       DatasetList list = new DatasetList();
       return list;
    }
+
+   private void transform(Run run, String role, String token) {
+      if (run == null || role == null) {
+         log.errorf("Transformation parameters error: run %s, role %s", run, role);;
+      }
+
+      Collection<RunService.RunSchema> runSchemas = getSchemas(run, role, token);
+
+      try (@SuppressWarnings("unused") CloseMe h1 = roleManager.withRoles(em, role);
+            @SuppressWarnings("unused") CloseMe h2 = roleManager.withToken(em, token)) {
+
+         Test test = Test.findById(run.testid);
+         boolean matched = false;
+         List<JsonNode> transformedSingleNodeResults = new ArrayList<>();
+         JsonNode transformedArrayNodeResult = null;
+         Collection<DataSet> dataSets = new ArrayList<>();
+         int ordinal = 0;
+         boolean functionError = false;
+         if (!runSchemas.isEmpty()){
+
+            Collection<Transformer> transformers = test.transformers;
+
+            for ( RunService.RunSchema s : runSchemas ){
+
+               if (!transformers.isEmpty()) {
+                  Map<Transformer, ObjectNode> extractorObjectsPerTrans = new HashMap<>();
+                  for (Transformer t : transformers) {
+
+                     ObjectNode root = JsonNodeFactory.instance.objectNode();
+                     List<Object[]> results = unchecked(em.createNamedQuery(QUERY_1ST_LEVEL_BY_RUNID_TRANSFORMERID_SCHEMA_ID) 
+                           .setParameter(1, run.id).setParameter(2, s.schemaid).setParameter(3, t.id)
+                           .unwrap(NativeQuery.class)
+                           .addScalar("name", TextType.INSTANCE)
```

Comment:
I just wonder - since you already use the named query in annotation, would it word if you provided the `SqlResultSetMapping` with appropriate `ColumnResult`s as well?

Reply:
I originally tried that. Hibernate ignores the ColumnResult object completely.